diff --git a/kohya_gui/class_folders.py b/kohya_gui/class_folders.py
index a0467fb..1cacc14 100644
--- a/kohya_gui/class_folders.py
+++ b/kohya_gui/class_folders.py
@@ -26,13 +26,13 @@ class Folders:

         # Set default directories if not provided
         self.current_output_dir = self.config.get(
-            "output_dir", os.path.join(scriptdir, "outputs")
+            "output_dir", os.path.join(".", "outputs")
         )
         self.current_logging_dir = self.config.get(
-            "logging_dir", os.path.join(scriptdir, "logs")
+            "logging_dir", os.path.join(".", "logs")
         )
         self.current_reg_data_dir = self.config.get(
-            "reg_data_dir", os.path.join(scriptdir, "reg")
+            "reg_data_dir", os.path.join(".", "reg")
         )

         # Create directories if they don't exist
