diff --git a/folder_paths.py b/folder_paths.py
index f0b3fd10..99cb613c 100644
--- a/folder_paths.py
+++ b/folder_paths.py
@@ -46,6 +46,9 @@ folder_names_and_paths["photomaker"] = ([os.path.join(models_dir, "photomaker")]
 
 folder_names_and_paths["classifiers"] = ([os.path.join(models_dir, "classifiers")], {""})
 
+folder_names_and_paths["wd14_tagger"] = ([os.path.join(models_dir, "wd14_tagger")], supported_pt_extensions)
+folder_names_and_paths["easy_use"] = ([os.path.join(models_dir, "easy_use")], supported_pt_extensions)
+
 output_directory = os.path.join(base_path, "output")
 temp_directory = os.path.join(base_path, "temp")
 input_directory = os.path.join(base_path, "input")
