From 94c5bb687b75270f54d37d34be24437c48c0ce39 Mon Sep 17 00:00:00 2001
From: "K." <kayo@illumium.org>
Date: Tue, 28 Jul 2026 16:21:50 +0300
Subject: [PATCH 2/2] Add command-line arg to set drivers path

---
 mik32_upload.py | 20 ++++++++++++++++----
 1 file changed, 16 insertions(+), 4 deletions(-)

diff --git a/mik32_upload.py b/mik32_upload.py
index 915c4ef..54b4227 100644
--- a/mik32_upload.py
+++ b/mik32_upload.py
@@ -213,6 +213,7 @@ def upload_file(
     post_action=default_post_action,
     mik_version=MIK32_Version.MIK32V2,
     use_driver=True,
+    drivers_path=default_drivers_path,
 ) -> int:
     """
     Запись прошивки в формате Intel HEX или бинарном в память MIK32.
@@ -220,6 +221,10 @@ def upload_file(
     @return: возвращает 0 в случае успеха, 1 - если прошивка неудачна
     """
 
+    drivers_build_path = (
+        default_drivers_build_path if drivers_path == default_drivers_path else ""
+    )
+
     print(f"Using {mik_version.value}")
 
     result = 0
@@ -298,9 +303,9 @@ def upload_file(
                     result |= eeprom.write_memory(
                         pages.pages_eeprom,
                         os.path.join(
-                            default_drivers_path,
+                            drivers_path,
                             "jtag-eeprom",
-                            default_drivers_build_path,
+                            drivers_build_path,
                             "firmware.hex",
                         ),
                     )
@@ -327,9 +332,9 @@ def upload_file(
                     result |= flash.write_pages_by_sectors(
                         pages.pages_spifi,
                         os.path.join(
-                            default_drivers_path,
+                            drivers_path,
                             "jtag-spifi",
-                            default_drivers_build_path,
+                            drivers_build_path,
                             "firmware.hex",
                         ),
                     )
@@ -495,6 +500,12 @@ def createParser():
         default=True,
         help="Отключает прошивку с использованием драйвера в ОЗУ",
     )
+    parser.add_argument(
+        "--drivers-path",
+        dest="drivers_path",
+        default=default_drivers_path,
+        help=f"Путь к файлам драйверов в ОЗУ для ускорения прошивки. По умолчанию: {default_drivers_path}",
+    )
     return parser
 
 
@@ -525,6 +536,7 @@ if __name__ == "__main__":
                 post_action=namespace.post_action,
                 mik_version=namespace.mcu_type,
                 use_driver=namespace.use_driver,
+                drivers_path=namespace.drivers_path,
             )
         )
     else:
-- 
2.54.0

